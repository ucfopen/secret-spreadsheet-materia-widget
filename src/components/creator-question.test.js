import React from 'react';
import Question from './creator-question';
import renderer from 'react-test-renderer';

describe('CreatorQuestion component', function() {

	beforeEach(() => {
		jest.resetModules()
	})

  test('CreatorQuestion renders question and description textareas with showQuestion = true', () => {
		const props = {
			qset: {
				question: 'What is love?',
				description: '',
			},
			showQuestion: true,
			questionRows: 1,
			descriptionRows: 1,
			handleQuestionChange: jest.fn(),
			handleDescriptionChange: jest.fn(),
		}

		const component = renderer.create(<Question {... props}/>);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();

	})

	test('CreatorQuestion renders question and description textareas with showQuestion = false', () => {
		const props = {
			qset: {
				question: 'What is love?',
				description: '',
			},
			showQuestion: false,
			questionRows: 1,
			descriptionRows: 1,
			handleQuestionChange: jest.fn(),
			handleDescriptionChange: jest.fn(),
		}

		const component = renderer.create(<Question {... props}/>);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();

  })

})
