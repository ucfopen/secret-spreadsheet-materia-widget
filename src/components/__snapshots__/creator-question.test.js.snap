// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreatorQuestion component CreatorQuestion renders question and description textareas with showQuestion = false 1`] = `
<div
  className="question-hidden"
>
  <div>
    <textarea
      className="question-text"
      onChange={[MockFunction]}
      placeholder="Question Text"
      rows={1}
      type="text"
      value="When September ends"
    />
  </div>
  <div>
    <textarea
      className="description-text"
      onChange={[MockFunction]}
      placeholder="Description Text"
      rows={1}
      type="text"
      value="CANT WAKE UP"
    />
  </div>
</div>
`;

exports[`CreatorQuestion component CreatorQuestion renders question and description textareas with showQuestion = true 1`] = `
<div
  className=""
>
  <div>
    <textarea
      className="question-text"
      onChange={[MockFunction]}
      placeholder="Question Text"
      rows={1}
      type="text"
      value="Wake me up"
    />
  </div>
  <div>
    <textarea
      className="description-text"
      onChange={[MockFunction]}
      placeholder="Description Text"
      rows={1}
      type="text"
      value="Wake me up inside"
    />
  </div>
</div>
`;
